engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -410,13 +437,15 @@ public void writeTuple(Object obj, String port)
       storage.writeDataItem(bos.toByteArray(), true);
       //logger.debug("Writing tuple for port id {}", pi.id);
       //fsOutput.hflush();
-      ++totalTupleCount;
       if (numSubscribers > 0) {
         // this is not asynchronous.  we need to fix this
         publishTupleData(pi.id, obj);
       }
-    } catch (IOException ex) {
-      logger.error(ex.toString());
+      ++totalTupleCount;
+    } catch (Exception ex) {
```

Comment:
IMO, it is necessary to handle RuntimeException that may be thrown by streamCodec.toByteArray() differently than IOExceptions from storage or wsClient. streamCodec.toByteArray() likely to throw an exception when they can't serialize tuples, while storage and wsClient unlikely to depend on tuple and will throw an exception for every tuple if the connection is lost or data can't be written to disk. The exception handling here corresponds more to handling of streamCodec.toByteArray().